test/OpenTelemetry.AutoInstrumentation.Tests/Configurations/FileBasedConfiguration/Parser/EnvVarTypeConverterTests.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using OpenTelemetry.AutoInstrumentation.Configurations.FileBasedConfiguration.Parser;
+using Xunit;
+using YamlParser = Vendors.YamlDotNet.Core.Parser;
+
+namespace OpenTelemetry.AutoInstrumentation.Tests.Configurations.FileBasedConfiguration.Parser;
+
+public class EnvVarTypeConverterTests
+{
+    private readonly EnvVarTypeConverter _converter = new();
+
+    [Fact]
+    public void ReadYaml_StringWithEnvVar_ReplacesCorrectly()
+    {
+        try
+        {
+            Environment.SetEnvironmentVariable("YAMLPARSER_TESTS_TEST_ENV", "HelloWorld");
+
+            var parser = new YamlParser(new StringReader("${YAMLPARSER_TESTS_TEST_ENV}"));
+            FileBasedTestHelper.MoveParserToScalar(parser);
+
+            var result = _converter.ReadYaml(parser, typeof(string), _ => null);
+
+            Assert.Equal("HelloWorld", result);
+        }
+        finally
+        {
+            Environment.SetEnvironmentVariable("YAMLPARSER_TESTS_TEST_ENV", null);
+        }
+    }
+
+    [Fact]
+    public void ReadYaml_WithFallback_UsesFallbackIfEnvNotSet()
+    {
+        var parser = new YamlParser(new StringReader("${YAMLPARSER_TESTS_NOT_EXIST:-FallbackValue}"));
+        FileBasedTestHelper.MoveParserToScalar(parser);
+
+        var result = _converter.ReadYaml(parser, typeof(string), _ => null);
+
+        Assert.Equal("FallbackValue", result);
+    }
+
+    [Fact]
+    public void ReadYaml_WithEnvVarPresent_IgnoresFallback()
+    {
+        try
+        {
+            Environment.SetEnvironmentVariable("YAMLPARSER_TESTS_TEST_ENV", "RealValue");
```

Comment:
I recommend using unique env variables for each test, just to eliminate sources of test flakiness